src/Tester/StreamingTests/PullingAgentManagementTests.cs

```diff
@@ -89,14 +90,16 @@ public static void MyClassCleanup()
 
             var states = await mgmt.SendControlCommandToProvider(adapterType, adapterName, (int)PersistentStreamProviderCommand.GetAgentsState);
             Assert.AreEqual(2, states.Length);
-            foreach (var state in states.Cast<PersistentStreamProviderState>())
+            foreach (var state in states)
             {
-                Assert.AreEqual(expectedState, state);
+                PersistentStreamProviderState providerState;
+                Enum.TryParse(state.ToString(), out providerState);
+                Assert.AreEqual(expectedState, providerState);
             }
 
             var numAgents = await mgmt.SendControlCommandToProvider(adapterType, adapterName, (int)PersistentStreamProviderCommand.GetNumberRunningAgents);
             Assert.AreEqual(2, numAgents.Length);
-            int totalNumAgents = numAgents.Cast<int>().Sum();
+            int totalNumAgents = numAgents.Select(n=>int.Parse(n.ToString())).Sum();
```

Comment:
Please remove those `n=>int.Parse(n.ToString()`.
We should not need to go via string.
If there is a problem with int to long, we can fix that.